server/anthropic/anthropic.go

```diff
@@ -144,6 +141,42 @@ func conversationToMessages(posts []llm.Post) (string, []anthropicSDK.MessagePar
 			}
 			currentBlocks = append(currentBlocks, imageBlock)
 		}
+
+		if len(post.ToolUse) > 0 {
+			for _, tool := range post.ToolUse {
+				toolBlock := anthropicSDK.ToolUseBlockParam{
+					ID:    anthropicSDK.F(tool.ID),
+					Type:  anthropicSDK.F(anthropicSDK.ToolUseBlockParamTypeToolUse),
+					Name:  anthropicSDK.F(tool.Name),
+					Input: anthropicSDK.Raw[any](tool.Arguments),
+				}
+				currentBlocks = append(currentBlocks, toolBlock)
+			}
+
+			resultBlocks := make([]anthropicSDK.ContentBlockParamUnion, 0, len(post.ToolUse))
+			for _, tool := range post.ToolUse {
+				if tool.Result != "" {
+					toolResultBlock := anthropicSDK.ToolResultBlockParam{
+						Type:      anthropicSDK.F(anthropicSDK.ToolResultBlockParamTypeToolResult),
+						ToolUseID: anthropicSDK.F(tool.ID),
+						Content: anthropicSDK.F([]anthropicSDK.ToolResultBlockParamContentUnion{
+							anthropicSDK.TextBlockParam{
+								Type: anthropicSDK.F(anthropicSDK.TextBlockParamTypeText),
+								Text: anthropicSDK.F(tool.Result),
+							},
+						}),
+					}
+					resultBlocks = append(resultBlocks, toolResultBlock)
+				}
+			}
+
+			if len(resultBlocks) > 0 {
+				flushCurrentMessage()
```

Comment:
Not related to this PR, but this pattern is hard to follow for me, I think I would prefer something like `flushMessage(currentRole, currentMessage)` for each call, making it very explicit. I fill this is kinda a "global variable" style. I don't know if there is any reason for doing it in this way.

Reply:
Agree, needs a refactor there. Will do in a follow up.